docs/design.md

```diff
@@ -117,37 +117,70 @@ All structs that are produced by the `ygen` library implement the `ygot.GoStruct
 
 For each enumerated entity (described above), an enumerated type in Go is
 generated, in a similar fashion to the `proto` library. Naming is according to
-the type of the enumerated leaf in YANG.
+the type of the enumerated leaf in YANG. Each name element is in camelcase, and
+when Go is generated, they are delimited by underscores in the same style as
+struct names.
 
 * `leaf` nodes with a type of `enumeration` are mapped to an enumeration named
   according to the path of the `leaf`. The path specified is
-  `ModuleName_LeafParentName_LeafName` such that a path of
-  `/interfaces/interface/state/enumerated-value` defined within the
+  `ModuleName_<PathElement1>_<PathElement2>_..._<PathElementN>_LeafName`, or
+  for compressed paths, `LeafGrandParentName_LeafName`, such that a
+  path of `/interfaces/interface/state/enumerated-value` defined within the
   `openconfig-interfaces` module is represented by an enumerated type named
-  `OpenconfigInterfaces_State_EnumeratedValue` (assuming path compression is
-  disabled), or `OpenconfigInterfaces_Interface_EnumeratedValue` when it is
-  enabled.
-  * This mapping is handled by `yang_helpers.go`:`resolveEnumName`.
+  `OpenconfigInterfaces_Interface_State_EnumeratedValue` (assuming path
```

Comment:
Is this actually `OpenconfigInterfaces_Interfaces_Interface_State_EnumeratedValue` or do we always skip PathElement1?